### PR TITLE
completes q12

### PR DESCRIPTION
**Pull request by HelloKt-8**
makes adjustments to the test, controller and model to account for comment_count based on article id.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -42,16 +42,16 @@ describe("GET /api/articles/:article_id", () => {
       .get("/api/articles/3")
       .expect(200)
       .then(({ body }) => {
-        const { articles } = body;
-        expect(articles).toMatchObject({
+        const { article } = body;
+        expect(article).toMatchObject({
           author: expect.any(String),
           title: expect.any(String),
           article_id: expect.any(Number),
```
Comment (LangaLee) [inline __tests__/app.test.js:49]: lets not test any here as we know the id 😄

File: __tests__/app.test.js [hunk 2]
```diff
@@ -42,16 +42,16 @@ describe("GET /api/articles/:article_id", () => {
...
-          body: expect.any(String),
           topic: expect.any(String),
           created_at: expect.any(String),
           votes: expect.any(Number),
           article_img_url: expect.any(String),
+          comment_count: expect.any(String)
```
Comment (MKD01) [inline __tests__/app.test.js:54]: This is a new behaviour so ideally this should have its own separate test.

**Review by LangaLee — COMMENTED** [review]
Nice work 😄

File: models/selectArticleId.models.js [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
\ No newline at end of file
+    return db.query(
+      `SELECT articles.article_id, articles.title, articles.topic, articles.author, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count
+      FROM articles
+      LEFT JOIN comments ON articles.article_id = comments.article_id
+      WHERE articles.article_id = $1
+      GROUP BY articles.article_id;`,
+      [article_id]
+    ).then(({ rows }) => {
+      if (rows.length === 0) {
```
Comment (MKD01) [inline models/selectArticleId.models.js:12]: Nice logic

**Review by MKD01 — COMMENTED** [review]
Great work

**Pull request merged**